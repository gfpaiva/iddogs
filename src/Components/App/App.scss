.app {
  &__content {
    min-height: calc(100vh - 200px);
    padding: 2rem 0;
  }
}
