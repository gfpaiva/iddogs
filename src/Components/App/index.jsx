import React from 'react';
import { Route, Switch } from 'react-router-dom';

import Header from '../Header';
import Footer from '../Footer';
import PrivateRoute from '../PrivateRoute';

import { PublicRoutes, PrivateRoutes } from '../../Utils/Routes';

import './App.scss';

export default function App() {
  return (
    <main className="app">
      <Header />
      <section className="app__content container">
        <Switch>
          {PrivateRoutes.map(route => (
            <PrivateRoute {...route} />
          ))}

          {PublicRoutes.map(route => (
            <Route {...route} />
          ))}
        </Switch>
      </section>
      <Footer />
    </main>
  );
}
