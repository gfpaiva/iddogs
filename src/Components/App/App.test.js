import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { shallow } from 'enzyme';

import App from './index';

describe('<App />', () => {
  it('should mount properly', () => {
    const wrapper = shallow(<App />);

    expect(wrapper).toMatchSnapshot();
  });
});
