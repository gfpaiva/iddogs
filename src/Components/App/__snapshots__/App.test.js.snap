// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should mount properly 1`] = `
<main
  className="app"
>
  <withRouter(Header) />
  <section
    className="app__content container"
  >
    <Switch>
      <PrivateRoute
        component={[Function]}
        exact={true}
        key="feed"
        path="/feed/:category/:id?"
      />
      <Route
        component={[Function]}
        exact={true}
        key="home"
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        key="signup"
        path="/signup"
      />
      <Route
        component={[Function]}
        exact={true}
        key="notfound"
      />
    </Switch>
  </section>
  <Footer />
</main>
`;
